geo/src/algorithm/convex_hull/mod.rs

```diff
@@ -138,7 +138,7 @@ where
 // Utility function: swap idx to head(0th position), remove
 // head (modifies the slice), and return head as a reference
 fn swap_remove_to_first<'a, T>(slice: &mut &'a mut [T], idx: usize) -> &'a mut T {
-    let tmp = std::mem::replace(slice, &mut []);
+    let tmp = std::mem::take(slice);
```

Comment:
Wow, does this actually work?

Reply:
Yes

Reply:
I think @lnicola's reaction is informative here: it's completely non-obvious what this call does now; you have to look at the docs (or your IDE) to figure out what it's doing compared to the existing call, and if there were no alternative I'd advocate for a comment.

Reply:
I don't mind it that much, I was just surprised that `&[T]` implements `Default`.

Reply:
That has been the case since 1.0

Reply:
from https://doc.rust-lang.org/std/mem/fn.take.html:

> **`take`**: Replaces dest with the default value of T, returning the previous dest value.

To my eye this change reads like pretty idiomatic usage of `take`. I'm struggling to think of what kind of clarifying comment we could add that isn't just the documentation for `take`, which seems like kind of a silly thing to do.

If anyone has a good comment suggestion, we could use it. Otherwise, I'd say it's fine as is.

Reply:
My point is that what the new code does is less obvious (to me) than the existing code, only because take's purpose is less immediately obvious than replace. I also realise this is far too much time to devote to such a minor change and nothing would ever get done if this was the norm – if you're both happy to accept it then I'm happy to go along with you.

Reply:
I went ahead and merged the PR, along with a [code comment](https://github.com/georust/geo/commit/d606fc5f01a6927f5e6033b09df7d404155b5637) to cover all the bases.